warp-academy-docs/docs/sdk/advanced/bundled-contract.md

```diff
@@ -30,79 +30,93 @@ the entry barrier, both for developers and given protocol users.
 
 ## How it works
 
-Instead of posting the contract and contract source transactions directly to Arweave mainnet, both are sent to Warp
-Gateway (`/gateway/contracts/deploy` endpoint) (this is the default behaviour of Warp's SDK `warp.createContract.deploy` function, when `forMainnet` instance is being used).
+### Nested bundles
 
-The Warp Gateway then:
+Bundling allows to write multiple data items into one top level transaction. A data item defers from a regular transaction by not allowing to transfer tokens and passing reward but it has most of the transaction's properties - such as owner, data, tags, owner and id.
 
-1. Posts contract transactions (i.e. the base contract transaction and contract source transaction - or only the
-   contract transaction, if deploying from existing source) to the Bundlr network. Each contract transaction is sent as
-   a separate Bundlr transaction - as the `data` of the bundled transaction. The Bundlr transaction in this case might
-   be considered as a "carrier" of the original transaction. Additionally - some additional tags to the Bundlr transaction
-   are added.
+In a nutshell, the nested bundles concept means that such data item of a bundle can also be a bundle containg other data items. According to ANS-104 standard it can lead to theoretically unbounded levels of nesting.
+
+You can read the specification for nested bundles standard in [ANS-104](https://github.com/ArweaveTeam/arweave-standards/blob/master/ans/ANS-104.md#31-nested-bundle).
+
+### Warp Gateway
+
+Instead of posting the contract and contract source transactions directly to Arweave mainnet, Warp creates data items from either both of them (when using `deploy` method) or only contract (when using `deployFromSourceTx` method) to Warp Gateway (`/gateway/v2/contracts/deploy` endpoint) (this is the default behaviour of Warp's SDK `warp.deploy` function, when `forMainnet` instance is being used).
```

Comment:
(this is the default behaviour of Warp's SDK `warp.deploy` function, when `forMainnet` instance is being used).

- as :::info